packages/appium/docs/en/ecosystem/build-drivers.md

```diff
@@ -2,4 +2,617 @@
+
+Basically, if you can't answer the question "how would I launch, remotely trigger behaviours, and
+read state from an app on this platform?" then you're not quite ready to write an Appium driver.
+Make sure you do the research to feel comfortable that there *is* a path forward. Once there is,
+coding it up and making it available as an Appium driver should be the easy part!
+
+## Other drivers to reference
+
+One of the greatest things about building an Appium driver is that there are already a number of
+open source Appium drivers which you can look at for reference. There is
+a [fake-driver](https://github.com/appium/appium/tree/2.0/packages/fake-driver) sample driver which
+does basically nothing other than showcase some of the things described in this guide.
+
+And of course, all of Appium's official drivers are open source and available in repositories at
+the project's GitHub organization. So if you ever find yourself asking, "how does a driver do X?",
+read the code for these drivers! Also don't be afraid to ask questions of the Appium developers if
+you get stuck; we're always happy to help make sure the driver development experience is a good
+one!
+
+## Basic requirements for Appium drivers
+
+These are the things your driver *must* do (or be), if you want it to be a valid Appium driver.
+
+### Node.js package with an Appium manifest
+
+All Appium drivers are fundamentally Node.js packages, and therefore must have a valid
+`package.json`. Your driver is not limited to Node.js, but whatever you do with your driver must
+ultimately be triggered by Node.js.
+
+Your `package.json` must include `appium` as a `peerDependency`. The requirements for the
+dependency versions should be as loose as possible (unless you happen to know your driver will only
+work with certain versions of Appium). For Appium 2.0, for example, this would look something like
+`^2.0.0`, declaring that your driver works with any version of Appium that starts with 2.x.
+
+Your `package.json` must contain an `appium` field, like this (we call this the 'Appium manifest'):
+
+    ```json
+    {
+        ...,
+        "appium": {
+            "driverName": "fake",
+            "automationName": "Fake",
+            "platformNames": [
+              "Fake"
+            ],
+            "mainClass": "FakeDriver"
+        },
+        ...
+    }
+    ```
+
+The required subfields are:
+
+* `driverName`: this should be a short name for your driver.
+* `automationName`: this should be the string users will use for their `appium:automationName`
+capability to tell Appium to use *your* driver.
+* `platformNames`: this is an array of platform names considered valid for your driver. When a user
+sends in the `platformName` capability to start a session, it must be included in this list for
+your driver to handle the session.
+* `mainClass`: this is a named export from your `main` field. It must be a class which extends
+Appium's `BaseDriver` (see below).
```

Comment:
```suggestion
* `mainClass`: this is a named export (in CommonJS style) from your `main` field. It must be a class which extends
Appium's `BaseDriver` (see below).
```

...just to avoid confusion with ESM named exports.  _Or_ we need to be explicit that Appium expects you to compile your ESM to CommonJS.  

We can support native ESM in the future, but that's the way things are now